src/db_queries/partners_dashboard_stats.ts

```diff
@@ -0,0 +1,114 @@
+import { Request, Response } from "express";
+import { PoolClient } from "pg";
+import { Conn } from "../db/conn";
+import { constants } from "../utils/constants";
+
+const pool = new Conn().pool;
+
+interface NgoStats {
+  /** total students who have enrolled in at least one course and who haven't cancelled it */
+  totalStudents: number;
+  /** total courses in which the students have enrolled (completed + not completed) */
+  totalCourses: number;
+  /** total hours corresponding to the lessons that haven't been cancelled in the courses (1h per lesson) */
+  totalHours: number;
+}
+
+async function getNgoStats(
+  client: PoolClient,
+  { orgId, year }: { orgId?: string; year?: number }
+): Promise<NgoStats> {
+  // We currently don't have orgId in courses table
+  // so need to use JOIN (course_student.student_id -> user.id)
+
+  const {
+    rows: [
+      {
+        total_students: totalStudents,
+        total_courses: totalCourses,
+        total_hours: totalHours,
+      },
+    ],
+  } = await client.query<{
+    total_students: number;
+    total_courses: number;
+    total_hours: number;
+  }>(`
+      SELECT
+        COUNT(DISTINCT ucs.student_id) total_students,
+        COUNT(DISTINCT ucs.course_id) total_courses,
+        COALESCE(SUM(CASE WHEN ct.duration = 3 THEN 14 WHEN ct.duration = 6 THEN 28 ELSE 0 END), 0) total_hours
```

Comment:
The number of hours should be calculated only until the current date and the cancelled lessons from the `users_courses_lessons_canceled` table should also be taken into consideration.

Reply:
Makes sense. I can do something like `ceiling(cancelled_or_now - start_date / 7)` to calculate weeks
So for 6 months, it will give 26 or 27 weeks (depending on how many days in months in the range). That works?

Reply:
Sounds good and we should also make sure that if the course was completed for example 2 months ago, the number of weeks for the course won't go over 14 (for 3-month courses) or 28 (for 6-month courses).